src/FixedGuns.cpp

```diff
@@ -194,12 +194,11 @@ bool FixedGuns::Fire(int num, Body *b)
 	m_temperature_stat[num] += m_gun[num].temp_heat_rate;
 	m_recharge_stat[num] = m_gun[num].recharge;
 
-	const matrix3x3d leadRot = m_shouldUseLeadCalc ? Quaterniond(m_currentLeadDir).ToMatrix3x3<double>() : matrix3x3d::Identity();
-
```

Comment:
I believe this has been fixed in my gun-manager rewrite branch already, but the construction of the quaternion here was definitely responsible for some part of the inaccuracy - I wasn't quite up to speed on quat semantics when I wrote the lead code originally :smile:

Reply:
I do not understand those quaternion transformations at all. Can you maybe point me to some documentation about those?

Reply:
There's precious little documentation in the Pioneer codebase, but I highly recommend referencing [EuclideanSpace](https://www.euclideanspace.com/maths/algebra/realNormedAlgebra/quaternions/) for a thorough understanding of quaternions at both a mathematic level and an applicational level. The issue in question was I was trying to construct a rotation quaternion from a normalized unit vector without any reference as to which axis should be pointing in that direction (nor the roll of the resulting rotation), which was not what the `Quaternion(vector3d)` constructor actually does. :smile:

Reply:
Thnx for the link!